docs/sql-reference/commands/select.md

```diff
@@ -661,7 +661,7 @@ When including multiple clauses, the same number of columns must be selected by
 
 The `ORDER BY` clause sorts a result set by one or more output expressions. `ORDER BY` is evaluated as the last step after any `GROUP BY` or `HAVING` clause. `ASC` and `DESC` determine whether results are sorted in ascending or descending order. When the clause contains multiple expressions, the result set is sorted according to the first expression. Then the second expression is applied to rows that have matching values from the first expression, and so on.
 
-The default null ordering is `NULLS LAST`, regardless of ascending or descending sort order.
+By default, NULL values sort as if larger than any non-null value; that is, NULLS FIRST is the default for DESC order, and NULLS LAST otherwise.
```

Comment:
I know this is pulled from the postgres [documentation](https://www.postgresql.org/docs/current/queries-order.html#:~:text=By%20default%2C%20null%20values%20sort,independently%20for%20each%20sort%20column.) but I think we can improve the legibility a bit by speaking or order (e.g., first, last) versus mechanism (e.g., larger, smaller). We might also want to add the ASC behavior explicitly like the postgres docs?

Perhaps something like this;

> By default, NULL values are sorted ahead of non-NULL values ('NULL FIRST') unless otherwise specified. With 'NULL LAST', NULL values are sorted after non-NULL values. 

NOTE: I'm a bit confused on the default order, so we should double check the functionality follows our documented behavior either way. Specifically, from the postgres docs, if a user specifies `ORDER BY`, `ASC` order is the default thus NULL values will be sorted `ASC` - and since `NULLS LAST` is the non-`DESC` default I'd assume NULLS would be sorted last by default BUT we also have "By default, null values sort as if larger than any non-null value" which seems to imply the opposite.

Reply:
Thanks for taking a look on this one Andrew. 
I agree with the suggested draft. 

As mentioned [here](https://firebolt-analytics.slack.com/archives/C02SMTT26VA/p1709561500187969?thread_ts=1709555190.448469&cid=C02SMTT26VA) about this behavior:
> It's the default in LQP2, no idea about LQP1. But it will definitely be the default in FB 2.0 

Adding @leonard-firebolt, @lorenzhs if there are additional insights.

Reply:
The behavior documented in PG matches our behavior and is IMO descriptive and precise enough. I would recommend to copy it and slightly change it to:

> The `NULLS FIRST` and `NULLS LAST` options can be used to determine whether nulls appear before or after non-null values in the sort ordering. By default, null values sort as if larger than any non-null value; that is, `NULLS FIRST` is the default for `DESC` order, and `NULLS LAST` is the default for `ASC` order.

Note that we will build a new, faster sort operator starting next month and we will revisit the documentation then again.

Reply:
Speaking of "larger" (defined in terms of the > operator) and "smaller" (defined in terms of the < operator) is IMO a very unambiguous way to describe sort order.